### DISCUSSION TITLE
Failed to create user (please aid)

### DISCUSSION THREAD

**Discussion by aaditya-cpu** [Budibase/budibase#2535] [Help]
unable to create admin user 
![image](https://user-images.githubusercontent.com/72136610/131968648-5c7c2838-2e23-422b-b816-7ed4d76d20ea.png)

This is my Yaml setup 
please aid 
`**``
#save in your root CMD mapped folder
#run 'docker images' before
#save the yaml and run the command : 'docker-compose up -d'

version: "3"

# optional ports are specified throughout for more advanced use cases.

services:
  app-service:
    restart: always
    image: budibase.docker.scarf.sh/budibase/apps
    container_name: bbapps
    ports:
      - "5005:4002"
    environment:
      SELF_HOSTED: 1
      COUCH_DB_URL: http://user:password@couchdb-service:5984
      WORKER_URL: http://worker-service:4003
      MINIO_URL: http://minio-service:9000
      MINIO_ACCESS_KEY: 987654
      MINIO_SECRET_KEY: MRS9879876
      INTERNAL_API_KEY: MRS_APIS_9879876
      BUDIBASE_ENVIRONMENT: 10
      PORT: 4002
      JWT_SECRET: MRS_JWT_9879876
      LOG_LEVEL: info
      SENTRY_DSN: https://a34ae347621946bf8acded18e5b7d4b8@o420233.ingest.sentry.io/5338131
      ENABLE_ANALYTICS: "true"
      REDIS_URL: redis-service:6379
      REDIS_PASSWORD: password
    volumes:
      - ./logs:/logs
    depends_on:
      - worker-service
      - redis-service

  worker-service:
    restart: always
    image: budibase.docker.scarf.sh/budibase/worker
    container_name: bbworker
    ports:
      - "5004:4003"
    environment:
      SELF_HOSTED: 1
      PORT: 4003
      CLUSTER_PORT: 999
      JWT_SECRET: MRS_JWT_9879876
      MINIO_ACCESS_KEY: MRS_MINIO_9879876
      MINIO_SECRET_KEY: MRS_MINIO_9879876
      MINIO_URL: http://minio-service:9000
      COUCH_DB_USERNAME: user
      COUCH_DB_PASSWORD: password
      COUCH_DB_URL: http://user:password@couchdb-service:5984
      INTERNAL_API_KEY: MRS_COUCH_9879876
      REDIS_URL: redis-service:6379
      REDIS_PASSWORD: password
    volumes:
      - ./logs:/logs
    depends_on:
      - redis-service
      - minio-service
      - couch-init

  minio-service:
    restart: always
    image: minio/minio
    volumes:
      - minio_data:/data
    ports:
      - "10000:9000"
    environment:
      MINIO_ACCESS_KEY: MRS_MINIO_9879876
      MINIO_SECRET_KEY: MRS_MINIO_9879876
      MINIO_BROWSER: "off"
    command: server /data
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost:9000/minio/health/live"]
      interval: 30s
      timeout: 20s
      retries: 3

  proxy-service:
    restart: always
    image: envoyproxy/envoy:v1.16-latest
    volumes:
      - ./envoy.yaml:/etc/envoy/envoy.yaml
    ports:
      - "8090:10000"
    depends_on:
      - minio-service
      - worker-service
      - app-service
      - couchdb-service

  couchdb-service:
    restart: always
    image: ibmcom/couchdb3
    environment:
      - COUCHDB_PASSWORD=password
      - COUCHDB_USER=user
    ports:
      - "3003:5984"
    volumes:
      - couchdb3_data:/opt/couchdb/data

  couch-init:
    image: curlimages/curl
    environment:
      PUT_CALL: "curl -u user:password -X PUT couchdb-service:5984"
    depends_on:
      - couchdb-service
    command: ["sh","-c","sleep 10 && $${PUT_CALL}/_users && $${PUT_CALL}/_replicator; fg;"]

  redis-service:
    restart: always
    image: redis
    command: redis-server --requirepass password
    ports:
      - "9000:6379"
    volumes:
      - redis_data:/data

  watchtower-service:
    image: containrrr/watchtower
    ports:
      - "7000:8080"
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
    command: --debug --http-api-update bbapps bbworker
    environment:
      - WATCHTOWER_HTTP_API=true
      - WATCHTOWER_HTTP_API_TOKEN=budibase
      - WATCHTOWER_CLEANUP=true
    labels:
      - "com.centurylinklabs.watchtower.enable=false"
volumes:
  couchdb3_data:
    driver: local
  minio_data:
    driver: local
  redis_data:
    driver: local

```**

**Comment by shogunpurple** [comment]
@aaditya-cpu since you have updated all of your ports, you will need to update your `envoy.yaml` to reflect these too. After updating, you need to restart that instance:
```
docker-compose restart proxy-service
```

**Reply by aaditya-cpu** [reply]
What command do i run to do this afte I make changes in yaml ? 
Where do I find this file ? in windows directory ?
Iam using CMD :

**Reply by shogunpurple** [reply]
it's in the same directory as your `docker-compose.yaml`

**Reply by aaditya-cpu** [reply]
So I just place it there and re run the command docker-compose up -d
Thanks for your help im just a noob trying to figure this setup for once

**Reply by shogunpurple** [reply]
Did you follow this guide in the docs?

https://docs.budibase.com/self-hosting/hosting-methods/docker-compose#setting-up-budibase

**Reply by aaditya-cpu** [reply]
Yes that How i arrived at this , and this one service is messing me

**Reply by shogunpurple** [reply]
so if you followed the guide, you should have the following files in your directory:

`envoy.yaml`
`hosting.properties`
`docker-compose.yaml`

- change your `envoy.yaml` to reflect your port changes in `docker-compose.yaml`
- `docker-compose restart proxy-service`

**Reply by aaditya-cpu** [reply]
ok, i see just have to place these files in directory , i was running around thinking if I had to run a command to create or boot them

**Reply by aaditya-cpu** [reply]
Error response from daemon: OCI runtime create failed: container_linux.go:380: starting container process caused: process_linux.go:545: container init caused: rootfs_linux.go:76: mounting "/run/desktop/mnt/host/c/Users/dev/envoy.yaml" to rootfs at "/etc/envoy/envoy.yaml" caused: mount through procfd: not a directory: unknown: Are you trying to mount a directory onto a file (or vice-versa)? Check if the specified host path exists and is the expected type
Still the same

**Reply by aaditya-cpu** [reply]
![image](https://user-images.githubusercontent.com/72136610/131977710-67d90ffd-9842-4ea4-913f-96b8d48d9b7c.png)

**Reply by shogunpurple** [reply]
From this error, it looks like your `envoy.yaml` file does not exist in:

```
/run/desktop/mnt/host/c/Users/dev/
```

Can you run an `ls` in that directory and show me the output?

**Reply by aaditya-cpu** [reply]
![image](https://user-images.githubusercontent.com/72136610/131977777-c8a5611e-a8be-4958-9be2-587d171c534e.png)
The containers have started but still cant assign a user

**Reply by aaditya-cpu** [reply]
```
 Directory of C:\Users\dev

03/09/2021  02:13 PM    <DIR>          .
03/09/2021  02:13 PM    <DIR>          ..
26/08/2019  12:45 PM    <DIR>          .android
05/06/2019  05:45 PM    <DIR>          .AndroidStudio3.1
03/09/2021  10:23 AM    <DIR>          .atom
22/04/2021  08:25 PM    <DIR>          .atom-backup
17/07/2021  12:36 PM               158 .bash_history
02/09/2021  05:49 PM                 2 .budibase.json
17/06/2021  02:42 PM    <DIR>          .conda
17/06/2021  02:40 PM                43 .condarc
21/04/2021  03:14 AM    <DIR>          .continuum
27/03/2021  01:12 PM    <DIR>          .dbus-keyrings
25/08/2021  01:14 PM    <DIR>          .docker
05/06/2019  05:39 PM    <DIR>          .dotnet
10/07/2021  04:11 PM                53 .git-for-windows-updater
25/08/2021  01:33 PM               261 .htaccess
23/03/2021  02:14 PM    <DIR>          .idlerc
07/06/2021  03:15 PM    <DIR>          .ipynb_checkpoints
13/04/2021  05:35 PM    <DIR>          .ipython
14/04/2021  11:43 AM    <DIR>          .jupyter
17/08/2021  03:51 PM    <DIR>          .kube
09/07/2021  12:45 PM    <DIR>          .linkassistant
09/07/2021  12:45 PM            10,268 .linkassistant.properties
13/06/2020  07:06 PM    <DIR>          .nbi
02/07/2021  03:05 PM                30 .node_repl_history
11/02/2021  09:17 AM    <DIR>          .ranktracker
11/02/2021  09:17 AM            17,205 .ranktracker.properties
25/08/2021  12:18 PM               129 .seopowersuite.properties
09/07/2021  10:44 AM    <DIR>          .seospyglass
09/07/2021  10:44 AM            16,343 .spyglass.properties
07/06/2019  03:47 PM    <DIR>          .templateengine
29/06/2021  07:11 PM    <DIR>          .VirtualBox
25/08/2021  12:51 PM    <DIR>          .websiteauditor
25/08/2021  12:51 PM            42,679 .websiteauditor.properties
12/06/2020  03:43 PM    <DIR>          .wp-cli
18/08/2021  03:50 PM    <DIR>          3D Objects
18/08/2021  03:50 PM    <DIR>          Contacts
21/08/2021  11:10 AM    <DIR>          Creative Cloud Files
02/09/2021  05:47 PM    <DIR>          Desktop
03/09/2021  01:09 PM             3,802 docker-compose.yaml
02/09/2021  07:01 PM    <DIR>          Documents
03/09/2021  01:46 PM    <DIR>          Downloads
03/09/2021  01:34 PM             4,368 envoy.yaml
18/08/2021  06:15 PM    <DIR>          Favorites
08/06/2019  01:54 PM             1,073 fibo.class
08/06/2019  02:05 PM               706 fibo.java
17/08/2021  01:15 PM    <DIR>          getting-started
18/08/2021  02:55 PM    <DIR>          Google Drive
03/09/2021  02:08 PM               610 hosting.properties
06/02/2020  12:03 PM               405 index.php
12/02/2020  05:24 PM            19,915 license.txt
18/08/2021  03:50 PM    <DIR>          Links
03/07/2021  10:17 AM    <DIR>          Local Sites
03/09/2021  11:17 AM    <DIR>          logs
07/06/2021  04:00 PM             1,389 Main-scrip.ipynb
18/08/2021  03:50 PM    <DIR>          Music
21/04/2021  03:43 AM    <DIR>          myenv
03/09/2021  09:20 AM    <DIR>          OneDrive
24/08/2021  04:33 PM    <DIR>          OpenVPN
28/08/2021  10:06 AM    <DIR>          Pictures
09/02/2021  05:36 PM                 0 python
26/06/2020  07:28 PM             7,278 readme.html
18/08/2021  03:50 PM    <DIR>          Saved Games
17/06/2021  02:41 PM             7,729 Scrape1.ipynb
18/08/2021  03:51 PM    <DIR>          Searches
19/06/2021  01:09 PM    <DIR>          Sitemap
05/06/2019  05:49 PM    <DIR>          source
14/05/2021  10:01 AM             4,068 The Random game.ipynb
19/04/2021  07:39 AM                72 Untitled.ipynb
21/04/2021  03:44 AM    <DIR>          VENV
18/08/2021  03:50 PM    <DIR>          Videos
28/07/2020  10:50 PM             7,101 wp-activate.php
22/02/2021  08:40 PM    <DIR>          wp-admin
06/02/2020  12:03 PM               351 wp-blog-header.php
09/10/2020  02:45 AM             2,328 wp-comments-post.php
03/09/2021  12:12 PM             2,823 wp-config-sample.php
03/09/2021  12:12 PM             3,199 wp-config.php
28/08/2021  05:56 PM    <DIR>          wp-content
31/07/2020  12:44 AM             3,939 wp-cron.php
22/02/2021  08:40 PM    <DIR>          wp-includes
06/02/2020  12:03 PM             2,496 wp-links-opml.php
06/02/2020  12:03 PM             3,300 wp-load.php
09/11/2020  04:23 PM            49,831 wp-login.php
14/04/2020  05:02 PM             8,509 wp-mail.php
12/11/2020  08:13 PM            20,975 wp-settings.php
01/10/2020  03:24 AM            31,337 wp-signup.php
09/10/2020  02:45 AM             4,747 wp-trackback.php
09/06/2020  01:25 AM             3,236 xmlrpc.php
25/08/2021  09:36 AM             1,110 _readme.txt
              39 File(s)        283,868 bytes
              50 Dir(s)  29,537,902,592 bytes free

```

**Reply by shogunpurple** [reply]
are you getting any errors in the browser console when you try to create your admin user? That should tell us more about what exactly is happening.

**Reply by aaditya-cpu** [reply]
![image](https://user-images.githubusercontent.com/72136610/131980631-f6b0ca13-65cb-48d3-b75f-de6dca609faa.png)

**Reply by aaditya-cpu** [reply]
http://localhost:5005/builder/assets/index.badf8004.js

**Reply by shogunpurple** [reply]
What does the network response say?

**Reply by aaditya-cpu** [reply]
![image](https://user-images.githubusercontent.com/72136610/131994957-80c81f7a-6d89-48d1-b36d-f7012e8e7a40.png)
![image](https://user-images.githubusercontent.com/72136610/131995075-6d775a7e-e12e-4524-80ca-e89c8662c73c.png)

**Reply by shogunpurple** [reply]
Ah, the problem is that you are going through your server port. Access your budibase installation here through your proxy service:

`localhost:8090`

**Reply by aaditya-cpu** [reply]
It worked, 
Whats the difference though ?

**Reply by aaditya-cpu** [reply]
Cookie “budibase:auth” will be soon rejected because it has the “SameSite” attribute set to “None” or an invalid value, without the “secure” attribute. To know more about the “SameSite“ attribute, read https://developer.mozilla.org/docs/Web/HTTP/Headers/Set-Cookie/SameSite index.badf8004.js:215:123709
Error fetching component definitions for app_dev_b5148696681b417bac97af8d54bb397c DOMException: The operation was aborted. index.badf8004.js:215:113260

​
Something new
and it keeps refreshing

**Reply by shogunpurple** [reply]
> It worked,
> Whats the difference though ?

The proxy service is the front gateway for budibase, so it routes requests properly. By hitting the server, it bypasses this, and you get 404s because it doesn't know how to route requests to the other services.


Ok, this looks like a browser thing. Can you try in a few different browsers, and if it still doesn't work, could you post the network request output?

**Reply by aaditya-cpu** [reply]
index.badf8004.js:215 GET http://localhost:8090/api/app_dev_b5148696681b417bac97af8d54bb397c/components/definitions 403 (Forbidden)
(anonymous) @ index.badf8004.js:215
(anonymous) @ index.badf8004.js:215
initialise @ index.badf8004.js:215
(anonymous) @ index.badf8004.js:215
async function (async)
(anonymous) @ index.badf8004.js:215
O1 @ index.badf8004.js:215
de @ vendor.7177f366.js:1
N1 @ index.badf8004.js:215
sa @ vendor.7177f366.js:1
b @ vendor.7177f366.js:1
Me @ vendor.7177f366.js:1
de @ vendor.7177f366.js:1
He @ vendor.7177f366.js:1
ca @ vendor.7177f366.js:1
ia @ vendor.7177f366.js:1
ua @ vendor.7177f366.js:1
de @ vendor.7177f366.js:1
pa @ vendor.7177f366.js:1
oa @ vendor.7177f366.js:1
la @ vendor.7177f366.js:1
b @ vendor.7177f366.js:1
I1 @ index.badf8004.js:215
D1 @ index.badf8004.js:215
de @ vendor.7177f366.js:1
M1 @ index.badf8004.js:215
sa @ vendor.7177f366.js:1
b @ vendor.7177f366.js:1
Me @ vendor.7177f366.js:1
de @ vendor.7177f366.js:1
He @ vendor.7177f366.js:1
ca @ vendor.7177f366.js:1
ia @ vendor.7177f366.js:1
ua @ vendor.7177f366.js:1
de @ vendor.7177f366.js:1
pa @ vendor.7177f366.js:1
oa @ vendor.7177f366.js:1
la @ vendor.7177f366.js:1
b @ vendor.7177f366.js:1
o7 @ index.badf8004.js:215
p @ index.badf8004.js:215
jt @ vendor.7177f366.js:1
Ft @ vendor.7177f366.js:1
Promise.then (async)
Ct @ vendor.7177f366.js:1
(anonymous) @ vendor.7177f366.js:1
(anonymous) @ vendor.7177f366.js:1
(anonymous) @ index.badf8004.js:215
async function (async)
(anonymous) @ index.badf8004.js:215
u @ vendor.7177f366.js:1
(anonymous) @ vendor.7177f366.js:1
Ft @ vendor.7177f366.js:1
Promise.then (async)
Ct @ vendor.7177f366.js:1
(anonymous) @ vendor.7177f366.js:1
(anonymous) @ vendor.7177f366.js:1
s @ vendor.7177f366.js:1
(anonymous) @ vendor.7177f366.js:1
Promise.then (async)
r @ vendor.7177f366.js:1
(anonymous) @ vendor.7177f366.js:1
setTimeout (async)
t.$$.update @ vendor.7177f366.js:1
de @ vendor.7177f366.js:1
za @ vendor.7177f366.js:1
_7 @ index.badf8004.js:215
de @ vendor.7177f366.js:1
(anonymous) @ index.badf8004.js:215
(anonymous) @ index.badf8004.js:215
Show 12 more frames

This is the har file -->https://drive.google.com/file/d/1RrRmMBFFOdiLhKhmuFo0DvPk4RIOWwMQ/view?usp=sharing
 
https://toolbox.googleapps.com/apps/har_analyzer/

**Reply by aaditya-cpu** [reply]
> 
> 
> > It worked,
> > Whats the difference though ?
> 
> The proxy service is the front gateway for budibase, so it routes requests properly. By hitting the server, it bypasses this, and you get 404s because it doesn't know how to route requests to the other services.
> 
> Ok, this looks like a browser thing. Can you try in a few different browsers, and if it still doesn't work, could you post the network request output?

Please check the file attached above

**Reply by shogunpurple** [reply]
It says that file does not exist, could you send it again?

**Reply by aaditya-cpu** [reply]
![image](https://user-images.githubusercontent.com/72136610/132299048-dcca321c-1eb7-4f4d-ba8d-5aeaaf92657c.png)
I am unable to acess the proxy server
https://pastebin.com/QVzKzZ3u    <-- Docker-compose
https://pastebin.com/V7tCp5kZ <---envoy
https://pastebin.com/ezaLi2Cr  <--0 hosting 

<html><body>
<span class="message-body-wrapper"><div class="url-preview"><div class="properties-view"><div class="tree-container">

GET | http://localhost:9093/editor/admin
-- | --


</span>
</body>
</html>
`The character encoding of the plain text document was not declared. The document will render with garbled text in some browser configurations if the document contains characters from outside the US-ASCII range. The character encoding of the file needs to be declared in the transfer protocol or file needs to use a byte order mark as an encoding signature.`

**Reply by aaditya-cpu** [reply]
I am able to access it from server side though

**Reply by aaditya-cpu** [reply]
> 
> 
> It says that file does not exist, could you send it again?

Awaiting your reply

**Reply by shogunpurple** [reply]
@aaditya-cpu your setup is completely misconfigured.

- your `CLUSTER_PORT` is wrong in your `docker-compose.yaml`, it should be pointing at your proxy service.  (9093) in your compose file
- all of your ports are wrong in `envoy.yaml`. The proxy service runs inside the docker network so you should use the internal ports of your docker containers
- Your `.env` file is basically pointless now, because you have hard coded all the ports into your `docker-compose.yaml`.

The reason we use the `.env` file is so that you can configure your ports from there, rather than hard-coding into the docker-compose.yaml.

**Reply by aaditya-cpu** [reply]
@shogunpurple  Thanks for your last response , I was able to redo the complete installation, however I wanted to know If you have any comprehensive tutorial of the product. As you can see there are some minor issues like populating the data from the data base is not pretty straightforward, Is there any video tutorial I can follow , I have checked your YouTube channel but not much help :(

**Reply by mjashanks** [reply]
All the docs that we have are on https://docs.budibase.com 

We do have a lot of work to do on them, which we are working on at the moment. 

If you have questions, please start another discussion!
